src/vs/editor/browser/widget/diffEditor/diffEditorWidget.ts

```diff
@@ -585,9 +585,15 @@ export class DiffEditorWidget extends DelegatingEditor implements IDiffEditor {
 
 		let diff: DiffMapping | undefined;
 		if (target === 'next') {
-			diff = diffs.find(d => d.lineRangeMapping.modified.startLineNumber > curLineNumber) ?? diffs[0];
+			diff = diffs.find(
+				d => d.lineRangeMapping.modified.startLineNumber > curLineNumber &&
+					d.lineRangeMapping.modified.endLineNumberExclusive !== d.lineRangeMapping.modified.startLineNumber
+			) ?? diffs[0];
```

Comment:
```
const modifiedLineCount = this._editors.modified.getModel()!.getLineCount();
if (modifiedLineCount === curLineNumber)
  diff = diffs[0];
else
  diff = diffs.find(d => d.lineRangeMapping.modified.startLineNumber > curLineNumber) ?? diffs[0];

I don't know if this will be better.

Reply:
> ```
> const modifiedLineCount = this._editors.modified.getModel()!.getLineCount();
> if (modifiedLineCount === curLineNumber)
>   diff = diffs[0];
> else
>   diff = diffs.find(d => d.lineRangeMapping.modified.startLineNumber > curLineNumber) ?? diffs[0];
> 
> I don't know if this will be better.
> ```
This seems to be a better solution. I have updated it.

This solution also solves the [following issue](https://github.com/microsoft/vscode/pull/219321#issuecomment-2199369440)